app/utils.py

```diff
@@ -898,6 +901,19 @@ def __init__(self, request, response, env):
         if self.auth and not self.auth.is_valid:
             self.auth = None
 
+        if self.admin_required:
```

Comment:
Can you comment why you handle it in Python code instead of relying on login: admin in app.yaml?